### DISCUSSION TITLE
Recommendations across files

### DISCUSSION THREAD

**Discussion by busti** [community/community#7125] [Copilot Conversations]
It feels like copilot isn't very good at recommending code related to that found in other files.  
I have a project where I use the same pattern in several files and it always recommends the wrong variable names, usually taking from what I have written before, but not from what I have written in other files.  
I.e. I have a lot of type definitions that look like this in one file:  
```scala
// Rx.scala
trait Rx[-A] { self =>
  type F[+_]
  type EffectT[FF[_]] = TransformK[FF, F]

  type G[-_]
  type StreamT[GG[_]] = TransformK[GG, G]

  type I[-T] <: Rx[T]
  type J[+T] <: Tx[T] { type J[+X] <: self.J[X] }
  
  // ... //
}

trait RxBuilder { self =>
  type G[-_]
  type StreamT[GG[_]] = TransformK[GG, G]

  type I[-T] <: Rx[T] { type G[-X] <: self.G[X] }

  // ... //
}
```

and like this in another:  
```scala
// Tx.scala
trait Tx[+A] { self =>
  type F[+_]
  type EffectT[FF[_]] = TransformK[FF, F]

  type H[+_]
  type StreamT[HH[_]] = TransformK[HH, H]

  type I[-T] <: Rx[T] { type I[-X] <: self.I[X] }
  type J[+T] <: Tx[T]
  
  // ... //
}  
```

and I am currently writing a second trait in the `Tx.scala`  file:  
```scala
trait TxBuilder { self =>
  type H[+_]
  type StreamT[HH[_]] = TransformK[HH, H]

  type J[+T] <: Tx[T] { /* recommendation goes here */ }
}
```

I would expect it to recommend the following `type H[+X] = self.H[X]` but instead it recommends `type I[-X] <: self.I[X]` probably because it has seen that pattern before in the same file.  

Admittedly, this is a very complex example, but I have seen the same show up in less complex stuff (variable names and such), I just couldn't easily isolate those.  
At this point I am not quite sure if it works across files like that, maybe my small example just lacks the data to make the completions smarter.

**Comment by enapt** [comment]
AFAIK, Copilot only works on single file recommendations at the current time.
